### ISSUE TITLE
Question about training

### ISSUE THREAD

**Issue by pepi99** [yangheng95/PyABSA#163]
I annotated my dataset (I have 1K positive, 1K negative and 1K neutral). 
My code is: 

```
warnings.filterwarnings('ignore')

seeds = [random.randint(0, 10000) for _ in range(2)]

apc_config_english = APCConfigManager.get_apc_config_english()
apc_config_english.model = APCModelList.FAST_LSA_T
apc_config_english.lcf = 'cdw'
apc_config_english.similarity_threshold = 1
apc_config_english.max_seq_len = 130
apc_config_english.dropout = 0
apc_config_english.cache_dataset = False
apc_config_english.patience = 10
apc_config_english.optimizer = 'radam'
apc_config_english.pretrained_bert = 'microsoft/deberta-v3-large'
apc_config_english.hidden_dim = 1024
apc_config_english.embed_dim = 1024
apc_config_english.num_epoch = 30
apc_config_english.log_step = 50
apc_config_english.learning_rate = 1e-5
apc_config_english.batch_size = 16
apc_config_english.evaluate_begin = 0
apc_config_english.l2reg = 1e-8
apc_config_english.seed = seeds

apc_config_english.cross_validate_fold = -1  # disable cross_validate

CryptoDataset = ABSADatasetList.CryptoCustom

sent_classifier = Trainer(config=apc_config_english,
                          dataset=CryptoDataset,  # train set and test set will be automatically detected
                          checkpoint_save_mode=1,  # =None to avoid save model
                          auto_device=True,  # automatic choose CUDA or CPU
                          path_to_save='/home/ubuntu/nlp/PyABSA/checkpoints/btr_model_statedict_new'
                          ).load_trained_model()

```

My training output (until now) looks as follows: 

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/45050191/160999218-a0dd95e7-5ca1-49f8-836e-1bd2eebbce44.png">

It seems strange to me that the accuracy at Epoch 4 is highest, then goes down, then goes up, and so on. Why doesn't it consistently increase with each epoch? Or, if gradients have reached a minimum, just stop changing? I don't understand why it goes up and down. Is this expected behaviour? 

Thanks.

**Comment by yangheng95** [comment]
Yes, you can set config.patience for early stop. e.g., config.patience=5

**Comment by pepi99** [comment]
ok thanks
